lua/lualine/config.lua

```diff
@@ -6,6 +6,265 @@ local modules = require('lualine_require').lazy_require {
   utils_notices = 'lualine.utils.notices',
 }
 
+---@alias LualineThemes
+---| "auto"
+---| "16color"
+---| "ayu_dark"
+---| "ayu_light"
+---| "ayu_mirage"
+---| "ayu"
+---| "codedark"
+---| "dracula"
+---| "everforest"
+---| "gruvbox_dark"
+---| "gruvbox_light"
+---| "gruvbox"
+---| "gruvbox-material"
+---| "horizon"
+---| "iceberg_dark"
+---| "iceberg_light"
+---| "iceberg"
+---| "jellybeans"
+---| "material"
+---| "modus-vivendi"
+---| "molokai"
+---| "moonfly"
+---| "nightfly"
+---| "nord"
+---| "OceanicNext"
+---| "onedark"
+---| "onelight"
+---| "palenight"
+---| "papercolor_dark"
+---| "papercolor_light"
+---| "PaperColor"
+---| "powerline"
+---| "powerline_dark"
+---| "pywal"
+---| "seoul256"
+---| "solarized_dark"
+---| "solarized_light"
+---| "Tomorrow"
+---| "wombat"
+
+---@class LualineDisabledFiletypes
+---only ignores the ft for statusline
+---@field statusline table
+---only ignores the ft for winbar.
+---@field winbar table
+
+---@class LualineRefresh
+---@field statusline integer
+---@field tabline integer
+---@field winbar integer
+
+---@class LualineOpts
+-- Show icons.
+---@field icons_enabled boolean
+-- Lualine theme name.
+-- Auto theme will load the theme automaically for your colorscheme
+-- or will generate one otherwise.
+---@field theme LualineThemes
+-- Separators between components.
+---@field component_separators table|string
+-- Separators between sections.
+---@field section_separators table|string
+-- Disable lualine for these filetypes.
+---@field disabled_filetypes LualineDisabledFiletypes
+-- When set to true, left sections i.e. 'a','b' and 'c'
+-- can't take over the entire statusline even
+-- if neither of 'x', 'y' or 'z' are present.
+---@field always_divide_middle boolean
+---@field globalstatus boolean
+---@field refresh LualineRefresh
+
+---@alias LualineComponents
+---| "branch"
+---| "buffers"
+---| "diagnostics"
+---| "diff"
+---| "encoding"
+---| "fileformat"
+---| "filename"
+---| "filesize"
+---| "filetype"
+---| "hostname"
+---| "location"
+---| "mode"
+---| "progress`"
+---| "tabs"
```

Comment:
You're mssing the "windows" components.

Also I think we should add these to this list too
```lua
---| function(self: table, is_focused: boolean): string
---| string
```

We can mark module here https://github.com/nvim-lualine/lualine.nvim/blob/5f68f070e4f7158517afc55f125a6f5ed1f7db47/lua/lualine/component.lua#L5 as a class and add that to this list too . In that case we can annotate self argument of fucntion as that class.